### PR TITLE
Improve typings and linter

### PR DESCRIPTION
**Pull request by marcoskwkm**
#### What problem is this solving?
This PR does mainly two things:
- Run `vtex setup`, so React typings are imported from other apps. By doing this, we don't need to redeclare the types that derive from our GraphQL;
- Improve linter rules and fix the resulting errors.

#### How should this be manually tested?

Interactions in [this workspace](https://types--checkoutio.myvtex.com/cart) should work fine.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable)](https://app.clubhouse.io/vtex/story/29027/melhorar-tipagem-e-linter-do-order-manager).
- [x] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
✔️ | Technical improvements

### CODE REVIEW

File: react/.eslintrc.json [hunk 1]
```diff
@@ -4,5 +4,8 @@
     "browser": true,
     "es6": true,
     "jest": true
+  },
+  "rules": {
+    "@typescript-eslint/no-empty-function": "off"
```
Comment (darlenedms) [inline react/.eslintrc.json]: Why was that needed?

Comment (marcoskwkm) [reply react/.eslintrc.json]: https://github.com/vtex-apps/checkout-cart/pull/41#discussion_r362068183

**Comment by jeffersontuc** [comment]
I tried to rebase this PR with `master`, but build started to fail. I think this might be related to what @victorhmp commented [here](https://github.com/vtex-apps/minicart/pull/207#issuecomment-579746212).
